vault/logical_system.go

```diff
@@ -2620,9 +2620,17 @@ func (b *SystemBackend) handleConfigUIHeadersUpdate(ctx context.Context, req *lo
 		return logical.ErrorResponse("X-Vault headers cannot be set"), logical.ErrInvalidRequest
 	}
 
+	if req.ResponseWriter == nil {
+		return logical.ErrorResponse("no ResponseWriter in the request"), logical.ErrInvalidRequest
+	}
+
 	// Translate the list of values to the valid header string
 	value := http.Header{}
 	for _, v := range values {
+		la := req.ResponseWriter.Header().Get("X-Vault-Listener-Add")
+		if b.Core.ExistCustomResponseHeader(header, la) {
```

Comment:
Shouldn't we refuse to allow setting a conflicting header for any listener, not just the one the update request came over on?

Reply:
Yes indeed!